### PR TITLE
Update rustdoc-internals.md

### PR DESCRIPTION
**Pull request by notriddle**
CC @GuillaumeGomez @camelid

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
It was a much needed update, thanks a lot!

**Review by tshepang — APPROVED** [review]

File: src/rustdoc-internals.md [hunk 1]
```diff
@@ -92,44 +102,66 @@ Here is the list of passes as of <!-- date-check --> November 2022:
 
 - `propagate-doc-cfg` propagates `#[doc(cfg(...))]` to child items.
 
+- `run-lints` runs some of rustdoc's lints, defind in `passes/lint`. This is
```
Comment (tshepang) [inline src/rustdoc-internals.md:105]:
```suggestion
- `run-lints` runs some of rustdoc's lints, defined in `passes/lint`. This is
```

File: src/rustdoc-internals.md [hunk 2]
```diff
@@ -92,44 +102,66 @@ Here is the list of passes as of <!-- date-check --> November 2022:
...
+  the last pass to run.
+
+  - `bare_urls` detects links that are not linkified, e.g., in Markdown such as
+    `Go to https://example.com/.` It suggests wrapping the link with angle brackets:
+    `Go to <https://example.com/>.` to linkify it. This is the code behind the <!--
+    date: 2022-05 --> `rustdoc::bare_urls` lint.
```
Comment (tshepang) [inline src/rustdoc-internals.md:111]:
```suggestion
    date-check: 2022-05 --> `rustdoc::bare_urls` lint.
```

File: src/rustdoc-internals.md [hunk 3]
```diff
@@ -92,44 +102,66 @@ Here is the list of passes as of <!-- date-check --> November 2022:
 
 - `propagate-doc-cfg` propagates `#[doc(cfg(...))]` to child items.
 
+- `run-lints` runs some of rustdoc's lints, defined in `passes/lint`. This is
+  the last pass to run.
+
+  - `bare_urls` detects links that are not linkified, e.g., in Markdown such as
+    `Go to https://example.com/.` It suggests wrapping the link with angle brackets:
+    `Go to <https://example.com/>.` to linkify it. This is the code behind the <!--
+    date-check: 2022-05 --> `rustdoc::bare_urls` lint.
```
Comment (tshepang) [inline src/rustdoc-internals.md:111]:
```suggestion
    date-check: may 2022 --> `rustdoc::bare_urls` lint.
```

File: src/rustdoc-internals.md [hunk 4]
```diff
@@ -148,23 +180,45 @@ to us"][video])
...
-crate, rustdoc will ask about trait implementations and generate new `Item`s
-for the impls based on that information. This goes directly into the returned
-`Crate` rather than roundabout through the `DocContext`. This way, these
-implementations can be collected alongside the others, right before rendering
-the HTML.
+It's important to note that rustdoc can ask the compiler for type information
+directly, even during HTML generation. This [didn't used to be the case], and
+a lot of rustdoc's architecture was designed around not doing that, but a
+`TyCtxt` is now passed to `formats::renderer::run_format`, which is used to
+run generation for both HTML and the (unstable as of March 2023) JSON format.
```
Comment (tshepang) [inline src/rustdoc-internals.md:187]:
```suggestion
run generation for both HTML and the
(unstable as of <!-- date-check --> March 2023) JSON format.
```

**Pull request merged**